src/modules/job-manager/plugins/perilog.c

```diff
@@ -1030,13 +1024,16 @@ static int proc_kill_timer_start (struct perilog_proc *proc, double timeout)
         proc->kill_timer = flux_timer_watcher_create (r,
                                                       timeout,
                                                       0.,
-                                                      proc_kill_timer_cb,
+                                                      proc_kill_timeout_cb,
                                                       proc);
         if (!proc->kill_timer) {
             flux_log_error (h,
                             "%s: failed to start %s kill timer",
                             idf58 (proc->id),
                             perilog_proc_name (proc));
+            /* Since timer cb won't be run, drain and send finish event now
+             */
+            proc_drain_and_finish (proc, false, true);
             return -1;
         }
```

Comment:
Suggestion: create the timer when proc is created but don't start it until here.  Then the error is not required.

Reply:
Deleted previous comment. I understand what you're saying now. Good idea.

Reply:
Actually let's leave this as is for now. It is only tangentially related to this PR, and the `remaining_time` member of the proc status response test for `proc->kill_timer` to determine if the timer is armed, so an alternate way to do that will need to be done (we never merged the change to expose if a watcher is "active").

The extra error handling doesn't seem like the end of the world, and there's a benefit of not creating the watcher when it isn't used 99% of the time.

But I'll handle this in a future PR. This one has gone on too long as it is.